core/deployment/src/main/java/io/quarkus/deployment/steps/RegisterForReflectionBuildStep.java

```diff
@@ -1,24 +1,32 @@
 package io.quarkus.deployment.steps;
 
+import java.util.function.Predicate;
+
 import javax.inject.Inject;
 
 import org.jboss.jandex.AnnotationInstance;
 import org.jboss.jandex.AnnotationValue;
 import org.jboss.jandex.ClassInfo;
 import org.jboss.jandex.DotName;
+import org.jboss.jandex.FieldInfo;
+import org.jboss.jandex.MethodInfo;
 import org.jboss.jandex.Type;
 import org.jboss.logging.Logger;
 
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
 import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveHierarchyBuildItem;
 import io.quarkus.runtime.annotations.RegisterForReflection;
 
 public class RegisterForReflectionBuildStep {
 
     private static final Logger log = Logger.getLogger(RegisterForReflectionBuildStep.class);
 
+    @Inject
+    BuildProducer<ReflectiveHierarchyBuildItem> reflectiveClassHierarchy;
```

Comment:
This can be part of the `@BuildStep` method.

Reply:
This change was addressed on current PR revision